overview.dd

```diff
@@ -757,6 +757,199 @@ long[] func(int x); // declares a function returning an array of longs
     )
 )
 
+
+$(SECTION3 Ranges,
+    $(P D uses the concept of a range in lieu of iterators or generators found in
+    other languages. A range is any type that provides a common interface to a
+    sequence of values. The purpose of a range is to allow for a simpler way to
+    write code that works on an arbitrary data, thereby making it reusable.
+    )
+
+    $(P The most basic type of range is called an input range,
+    which provides three methods.
+    )
+
+-----------------------
+struct MyRange
+{
+    auto front()
+    {
+        // return the next value in the sequence
+    }
+
+    void popFront()
+    {
+        // move the front of the sequence to the next value
+    }
+
+    bool empty()
+    {
+        // true if the range has no more values to return
+    }
+}
+-----------------------
+
+    $(P To understand the power of this simple interface, let's run through an
+    example. Say we wanted to write a program that took all of the employees in
+    a company, took out the ones younger than 40, and grouped the remaining into
+    and array of arrays by their organization.
+    )
+
+-----------------------
+struct Employee
+{
+    uint id;
+    uint organization_id;
+    string name;
+    uint age;
+}
+
+struct Employees
```

Comment:
What did Ali use in his book?

If this example is kept, at least the text should say near the beginning that this is an illustrative example. We don't want people to think that they need to write a range by hand just to map/filter an array.